source/style.rst

```diff
@@ -0,0 +1,171 @@
+
+One-liner docstrings:
+^^^^^^^^^^^^^^^^^^^^^
+`PEP-257`_ says "One-liners are for really obvious cases." In these cases, the
+quotes should be on the same line with no extra whitespace surrounding the
+docstring. Code should begin on the following line. For example:
+
+.. code:: python
+
+    class FeeFilter(InOutPositioner):
+        """A single attenuation blade, as implemented in the FEE."""
+        state = Cpt(EpicsSignal, ':STATE', write_pv=':CMD')
+
+Two-liner docstrings:
+^^^^^^^^^^^^^^^^^^^^^
+Sometimes, a class or function is simple enough to warrant a one-liner but the
+docstring won't fit within `PEP-8`_'s 79 character limit. In this case, the
+quotes may be moved to separate lines, as shown below:
+
+.. code:: python
+
+    def screen(self):
+        """
+        Opens Epics motor expert screen e.g. for reseting motor after stalling.
+        """
+        executable = 'motor-expert-screen'
+
+If the description still does not fit, you may use two lines:
+
+.. code:: python
+
+    def get_raw_mesh_voltage(self):
+        """
+        Get the current acromag voltage that outputs to the HV supply, i.e
+        the voltage seen by the HV supply.
+        """
+        return self.read_sig.get()
+
+Multi-line docstrings:
+^^^^^^^^^^^^^^^^^^^^^^
+When a short summary is not sufficient, multi-line docstrings must be used.
+In these cases, an additional description can be given following the summary,
+separated by a blank line. Furthermore, please embellish this documentation with
+sections like Parameters or Attributes as specified by the `numpydoc`_ standard.
+
+Some PCDS-specific rules for these docstrings:
+
+* Triple quotes should be on their own lines.
+* The docstring should be followed by a blank line.
+* The summary must be single line. If further explanation is necessary, move it
+  to the description, which can be as long as you want.
+* Class parameters should be described in the class's docstring under the
+  Parameters section. Therefore, the :meth:`__init__` method can be blank.
+
+Please note: In list sections, the colon must be preceded by a space, or omitted if the type is absent.
+
+.. code::
+
+    class BaseGon(Device, BaseInterface):
+        """
+        Basic goniometer, as present in XPP.
+
+        This requires eight motor PV prefixes to be passed in as keyword
+        arguments, and they are all labelled accordingly.
+
+        Parameters
+        ----------
+        name : str
+            A name to refer to the goniometer device.
+        ...
+        prefix_tilt : str
+            The EPICS base PV of the sample-stage's tilt motor.
+        """
+
+        hor = FCpt(IMS, '{self._prefix_hor}', kind='normal')
+
+Extra Notes
+-----------
+* Physical units should be specified in a parameter's description, not with its
+  type.
+* Inline code blocks can be specified with double-backticks
+  (e.g. ````return 0````.
+* Treat PV names as strings, surrounding them with single-quotes
+  (e.g. 'CXI:JET:X').
```

Comment:
Any reason to prefer single quotes here over double quotes?

Reply:
Not really, just picked one. I like single better and they seem more prevalent in the code I've worked with. If you have an argument for double quotes, I'm open to changing it.

Reply:
Nah, just the standard "I wonder if there's a reason for it". I've heard arguments for using all single quotes and for using all double quotes and none of them have really convinced me.

Reply:
My code is mostly `'` and `'''` - it means 1 (3) less times I have to press shift for every opening/closing string 😄 .

In my opinion, there should always be an "out" for using `'` or `"` that goes against the standard if the string itself contains `'` or `"` (escaping is much uglier).

Reply:
I agree with the out instead of escaping, much easier and prettier.
I stuck with `"""` for docstrings just to go along with PEP-257, but good reason 👌 😆